openmdao/core/system.py

```diff
@@ -3632,8 +3632,14 @@ def list_outputs(self,
             impl_outputs = {}
             if residuals_tol:
                 for n, m in outputs.items():
-                    if "resids" in m and n in states and m['resids'] > residuals_tol:
-                        impl_outputs[n] = m
+                    if "resids" in m and n in states:
+                        if len(m['resids']) > 1:
```

Comment:
Check to make sure that for a scalar variable, the value of m['resids'] ends up being a size 1 array, because calling len(x) where x is a scalar will fail.